### PR TITLE
Rename Deployment API structs

### PR DESCRIPTION
**Pull request by ghodss**
As per conversation with @nikhiljindal, tweak the names of some of the Deployment API structs.

### CODE REVIEW

**Comment by ncdc** [comment]
cc @ironcladlou

File: docs/proposals/deployment.md [hunk 1]
```diff
@@ -246,7 +248,7 @@ To begin with, we will support 2 types of deployment:
   This results in a slower deployment, but there is no downtime. At all times
   during the deployment, there are a few pods available (old or new). The number
   of available pods and when is a pod considered "available" can be configured
-  using RollingUpdateDeploymentSpec.
+  using RollingUpdateDeploymentStragegy.
```
Comment (ironcladlou) [inline docs/proposals/deployment.md]: Typo

**Comment by ironcladlou** [comment]
One small typo, otherwise lgtm.

File: docs/proposals/deployment.md (hunk 1 above)
Comment (ghodss) [inline docs/proposals/deployment.md]: Done

File: pkg/util/procfs/example_proc_cgroup [hunk 2]
```diff
@@ -7,4 +7,4 @@
 5:memory:/user/1000.user/c1.session
 4:cpuacct:/user/1000.user/c1.session
 3:cpu:/user/1000.user/c1.session
-2:cpuset:/
\ No newline at end of file
+2:cpuset:/
```
Comment (nikhiljindal) [inline pkg/util/procfs/example_proc_cgroup]: unintended?

**Comment by nikhiljindal** [comment]
Thanks! API renames LGTM.

An unrelated change seems to have creeped in.
Will wait for that to be removed, before applying the LGTM label.

**Comment by ghodss** [comment]
Unintended change removed

**Comment by nikhiljindal** [comment]
LGTM, thanks!

**Comment by ghodss** [comment]
@k8s-bot test this please

**Comment by nikhiljindal** [comment]
Jenkins failed to fetch git repo, retrying.
@k8s-bot ok to test

**Comment by ghodss** [comment]
@k8s-bot test this please

**Comment by ghodss** [comment]
@k8s-bot test this please

**Comment by nikhiljindal** [comment]
The last 5 Jenkins failures have been:
```
Unable to fetch git repo - Twice
```
```
Build timed out
```
```
Kubernetes e2e suite.Probing container with readiness probe should not be ready before initial delay and never restart
```
```
Kubernetes e2e suite.Daemon set should launch a daemon pod on every node of the cluster
Kubernetes e2e suite.Kubectl client Kubectl expose should create services for rc
```

@k8s-bot ok to test

**Comment by nikhiljindal** [comment]
Flaky test:
```
Kubernetes e2e suite.Probing container with readiness probe should not be ready before initial delay and never restart
```

@k8s-bot ok to test

**Comment by alex-mohr** [comment]
@k8s-bot test this again

**Comment by ghodss** [comment]
@k8s-bot test this please

**Pull request merged**